### PR TITLE
chore: Styles and Integrates Image Gallery

### PR DESCRIPTION
**Pull request by danzanzini**
## What's the purpose of this pull request?

This PR addresses the tasks of [Style](https://vtex-dev.atlassian.net/browse/FSSS-58) and [Integrate](https://vtex-dev.atlassian.net/browse/FSSS-55) the [ImageGallery Component](https://github.com/vtex-sites/base.store/blob/master/src/components/ui/ImageGallery/ImageGallery.tsx) to the Product Display Page, as shown in this [figma file](https://www.figma.com/file/zHMOtOPduiM0hNNcg21iR1/Handoff-PDP-v0.1?node-id=18%3A33225).

There are still some minor adjustments to make. I've added them to the checklist below and I may tackle it in this PR or in future PRs so we can merge this one.

## How does it work?

I've changed the ImageGallery to not use the [Faststore's useSlider](https://github.com/vtex/faststore/blob/master/packages/ui/src/hooks/useSlider/useSlider.ts) component in favor of an approach using the default browser scroll, with a button controlling it through javascript.

The results are as follows:
|State |Desktop  | Mobile|
--- | --- | ---|
|1 item|image|image|
|> 1 item|image|image|
|with scroll|image|image](https://user-images.githubusercontent.com/8497187/162510633-5659d2d3-07fb-4e74-9e49-4407519674f1.gif)|ezgif com-gif-maker(1)|

## How to test it?

Go to the preview link and test the PDP of products with different amounts of product images:
- [with only 1 image](https://preview-409--base.preview.vtex.app/aedle-vk1-headphone-99988211/p)
- [with more than 1 image, but without scroll](https://preview-409--base.preview.vtex.app/awesome-plastic-bacon-9938908/p)
- [with scroll](https://preview-409--base.preview.vtex.app/awesome-wooden-bike-sleek-47501742/p)

Remember to test it for desktop and mobile. Try around with the scroll, use it with the buttons or with the scroll bar.

## Checklist

<em>You may erase this after checking them all ;)</em>

- [x] CHANGELOG entry added
- [x] Remove padding of selector items when there is no scroll [Jira task created](https://vtex-dev.atlassian.net/browse/FSSS-280)
- [x] Display/hide arrow buttons according to scroll position [Jira task created](https://vtex-dev.atlassian.net/browse/FSSS-280)
- [x] Uses aspect ratio instead of fixed height to define max height of main image
- [x] Adding Zoom tool [Jira task created](https://vtex-dev.atlassian.net/browse/FSSS-281)
- [x] Themify component [Jira task created](https://vtex-dev.atlassian.net/browse/FSSS-282)

Thanks to @renatamottam @saranicoly @heitorado @filipeafns and @lucasfp13 for all the help

### CODE REVIEW

**Review by filipeafns — CHANGES_REQUESTED** [review]
Hey @danzanzini. This is a great work, congratulations 🎉 

I Have a single issue I found while testing this on both Chrome and Safari, Desktop and Mobile:

- I know this is similar to the design I did, but I think we should decrease the overlap between the buttons with the first and last button within the scroll area (check attachment 1) `data-fs-image-gallery-selector-elements`.

attachement 1

- Maybe this will solve the other the issue with the last thumb, that's hidden behind the button (check attachment 2)
attachement 2

**Review by renatamottam — APPROVED** [review]
Amazing job, @danzanzini 👏 👏 👏

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,11 +16,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Adds Storybook configs (#463)
 - Adds vtex search tracking script. With this we will populate TopSearches and Autocomplete indices (#389)
 - Add `RegionalizationBar`, `RegionalizationButton` components and integrates it on Mobile and Desktop devices (#424).
+- Integrates and styles ImageGallery component at the PDP (#409)
 
 ### Changed
 
 - Applies new local tokens to `BannerText` (#470)
 - Update the Incentives component to handle CMS data (#474)
+- Refactors ImageGallery component to not use the useSlider component (#409)
```
Comment (filipewl) [inline CHANGELOG.md:25]: Can you move these new entries to the beginning of their sections? [The latest version comes first.](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).

**Review by saranicoly — APPROVED** [review]
LGTM. Great work!

File: src/components/ui/ImageGallery/ImageGallerySelector.tsx [hunk 2]
```diff
@@ -1,121 +1,84 @@
...
+              key={idx}
+              data-thumbnail-button={
+                idx === currentImageIdx ? 'selected' : 'true'
+              }
+              aria-label={`Load ${image.alternateName} - Image ${idx + 1} of ${
+                images.length
+              }`}
+              onClick={() => onSelect(idx)}
+            >
+              <Image
```
Comment (filipewl) [inline src/components/ui/ImageGallery/ImageGallerySelector.tsx:64]: Is there a way to use responsive images here? LH complained about not using them.

CleanShot 2022-04-22 at 11 35 10

CleanShot 2022-04-22 at 11 35 29

You can check the report here https://googlechrome.github.io/lighthouse/viewer/ by pasting the JSON file inside [here](https://github.com/vtex-sites/base.store/files/8541838/sfj-443454d--base.preview.vtex.app_2022-04-22_11-27-59.json.zip).

File: src/components/ui/ImageGallery/ImageGallerySelector.tsx [hunk 3]
```diff
@@ -1,121 +1,84 @@
...
+      )}
+      <div data-fs-image-gallery-selector-elements ref={elementsRef}>
+        {images.map((image, idx) => {
+          return (
+            <Button
+              key={idx}
+              data-thumbnail-button={
+                idx === currentImageIdx ? 'selected' : 'true'
+              }
+              aria-label={`Load ${image.alternateName} - Image ${idx + 1} of ${
```
Comment (filipewl) [inline src/components/ui/ImageGallery/ImageGallerySelector.tsx:59]: Do we really need to mention "load"?
```suggestion
              aria-label={`${image.alternateName} - Image ${idx + 1} of ${
```

**Review by filipewl — COMMENTED** [review]
Great job @danzanzini (and @renatamottam?)!!!

I [found](https://github.com/vtex-sites/base.store/pull/409#discussion_r856299267) a Lighthouse's responsive images issue in the PDP version with the gallery & scroll. What do you think of updating the PDP's URL for LH and tests (in [store.config.js](https://github.com/vtex-sites/base.store/commit/443454d2077e9f1cbf9816c2c5c62b57d94e67bf)) to use that version to account for a more "complete" PDP version?

**Comment by filipewl** [comment]
Closed in favor of https://github.com/vtex-sites/gatsby.store/pull/6.

**Pull request closed without merging**